### PR TITLE
soc: x86: add Lakemont SoC

### PR DESCRIPTION
**Pull request by dcpleung**
This adds a very basic SoC configuration for Intel Lakemont SoC.

### CODE REVIEW

**Comment by dcpleung** [comment]
This also includes https://github.com/zephyrproject-rtos/zephyr/pull/31198. Will rebase once that is merged.

**Comment by dcpleung** [comment]
#31198 is merged. So this is ready for review.

**Review by jhedberg — APPROVED** [review]

File: soc/x86/lakemont/linker.ld [hunk 1]
```diff
@@ -0,0 +1,8 @@
+/*
+ * Copyright (c) 2011-2014, Wind River Systems, Inc.
+ *
```
Comment (jenmwms) [inline soc/x86/lakemont/linker.ld:3]: missing Copyright (c) 2021 Intel Corporation

Comment (dcpleung) [reply soc/x86/lakemont/linker.ld:3]: This is a straight copy from `soc/x86/atom/linker.ld` without any modifications, hence no new copyright line.

**Review by jenmwms — APPROVED** [review]

File: soc/x86/lakemont/linker.ld (hunk 1 above)
Comment (jenmwms) [reply soc/x86/lakemont/linker.ld:3]: Oh ok, thank you!

**Review by aasthagr — APPROVED** [review]

**Comment by dcpleung** [comment]
CI failure is related to https://github.com/zephyrproject-rtos/zephyr/issues/32344

**Comment by nashif** [comment]
can we have a user of this? maybe in qemu?

**Comment by dcpleung** [comment]
> can we have a user of this? maybe in qemu?

Added a new QEMU board in last push.

File: boards/x86/qemu_x86/qemu_x86_lakemont.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+name: QEMU Emulation for X86 (Lakemont)
+type: qemu
+simulation: qemu
+arch: x86
+toolchain:
+  - zephyr
+  - xtools
+  - llvm
+testing:
+  default: true
```
Comment (nashif) [inline boards/x86/qemu_x86/qemu_x86_lakemont.yaml:11]: make this for now just run the kernel tests

```
  only_tags:
     - kernel
```

Comment (dcpleung) [reply boards/x86/qemu_x86/qemu_x86_lakemont.yaml:11]: Added.

**Pull request merged**